src/main/java/org/nixos/idea/settings/NixLangSettings.java

```diff
@@ -0,0 +1,78 @@
+package org.nixos.idea.settings;
+
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.components.PersistentStateComponent;
+import com.intellij.openapi.components.RoamingType;
+import com.intellij.openapi.components.State;
+import com.intellij.openapi.components.Storage;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayDeque;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Deque;
+
+@State(name = "NixLangSettings", storages = @Storage(value = "nix-idea-tools.xml", roamingType = RoamingType.DISABLED))
+public final class NixLangSettings implements PersistentStateComponent<NixLangSettings.State> {
```

Comment:
polish: I would prefer if you keep this state class more specific for now. Like `NixExternalFormatterSettings`. Here are my two reasons:

1. I don't want to rule out that we may add a simple built-in formatter to the plugin at some point.
2. `nix-idea-tools.xml` is intended to contain configurations related to external dependencies, like installed applications. There may be other configurations in the future, which should not go into this file.

Reply:
for reason (1), if you add an non-external formatter, I think it would appropriate to add its settings to this class also (rather than the existing settings in the "Build" section)

for reason (2), if you prefer a different file to `nix-idea-tools.xml`, we can change that. I am not very familiar with how the `@State` and `@Storage` abstraction works, I just wanted a settings screen for this plugin under the "Languages" section.

New settings can go underneath like in the picture below:

<img width="652" alt="Screenshot 2024-06-03 at 21 18 54" src="https://github.com/NixOS/nix-idea/assets/45274424/38192a4a-696f-4612-885d-6858f8b65971">

If you still feel like the external formatter deserves its own section, I am happy to make another one too though! Again, no strong opinions from me on this

Reply:
My concern was mostly about where and how to store the configuration, not the UI. The configuration of a built-in formatter would be system-independent and can be synchronized across operating systems. External formatters should not be synchronized across operating systems. To make this separation, these two cases need to be stored in separate XML files.

The `nix-idea-tools.xml` is supposed to become the file storing these system-dependent configurations. This means your initial choice of that XML file was actually correct. Since this XML file will not contain other language-related configurations, I just don't want to call it `NixLangSettings`.

---

```xml
<application>
  <component name="NixLspSettings">
    <!-- LSP settings -->
  </component>
  <component name="NixExternalFormatterSettings">
    <!-- External formatter settings -->
  </component>
</application>
```

instead of

```xml
<application>
  <component name="NixLspSettings">
    <!-- LSP settings -->
  </component>
  <component name="NixLangSettings">
    <!-- Other system-dependent settings -->
  </component>
</application>
```

Reply:
I see, thanks for the explanation! That makes sense, I will apply your suggestion then.